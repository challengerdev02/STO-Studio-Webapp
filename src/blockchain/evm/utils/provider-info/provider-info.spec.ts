import { cleanup } from '@testing-library/react';
import { Web3ProvidersInfo } from './index';

describe('Test Web3ProvidersInfo object', () => {
  afterEach(cleanup);

  const setup = () => {
    return {
      ...Web3ProvidersInfo,
    };
  };

  it('should test Web3ProvidersInfo', function () {
    const res = setup();

    expect(res).toBeTruthy();
    expect(res['bsc'].check).toBe('isBinanceChainWallet');
    expect(res['bsc'].id).toBe('bsc');
    expect(res['fortmatic'].id).toBe('fortmatic');
    expect(res['fortmatic'].type).toBe('web');
    expect(res['authereum'].name).toEqual('Authereum');
  });

  it('should match with snapshot', () => {
    // @ts-ignore
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });
});
