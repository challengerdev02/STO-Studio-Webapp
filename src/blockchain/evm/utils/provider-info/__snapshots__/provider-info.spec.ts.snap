// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Web3ProvidersInfo object should match with snapshot 1`] = `undefined`;
